---
title: "MSK SPECTRUM data freeze: ectopic expression in cancer cell clusters"
author: "Florian Uhlitz"
date: "`r Sys.Date()`"
output: 
  html_document: 
    df_print: kable
    number_sections: yes
    toc: yes
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}

knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = F)

```

# Ectopic gene expression in cancer cell clusters

```{r chunk_340_010}

library(magrittr)
library(tidyverse)
library(Seurat)
library(readxl)
library(cowplot)
library(viridis)
library(ggpubr)
library(ggrepel)

```

```{r chunk_340_020}

## load global vars: 
source("src/global_vars.R")

# scrna_meta_tbl
# clrs
# markers_v7
# markers_v7_super
# cell_type_super_lookup

```

```{r chunk_340_040}

## load seurat objects with expression data
seu_obj <- read_rds("/work/shah/users/uhlitzf/data/SPECTRUM/freeze/v7/Ovarian.cancer.super_processed_filtered_annotated.rds")
marker_tbl <- read_tsv("/work/shah/users/uhlitzf/data/SPECTRUM/freeze/v7/Ovarian.cancer.super_ectopic_expression_markers.tsv")

```


```{r chunk_340_100}

marker_tbl_sub <- marker_tbl %>% 
  filter(pct.1 < 0.5 | pct.2 < 0.5) %>% 
  arrange(-abs(avg_log2FC)) %>% 
  filter(comparison == "Other_vs_Adnexa")

## genes of interest: 
## differentially expressed in Adnexa vs Non-Adnexa within a given cluster
gois_tbl <- marker_tbl_sub %>% 
  # filter(cluster_label %in% c("Cancer.cell.1", "Cancer.cell.2", "Cancer.cell.3")) %>%
  group_by(cluster_label) %>% 
  slice(1:50) %>% 
  ungroup() %>% 
  select(cluster_label, gene) %>% 
  mutate(filter_column = T)

gois <- gois_tbl %>% 
  pull(gene) %>% 
  unique()
  
my_features <- c("sample", "cluster_label", gois)

## expression data for genes of interest per single cell
plot_data <- FetchData(seu_obj, my_features) %>% 
  as_tibble() %>% 
  gather("gene", "value", -sample, -cluster_label) %>% 
  left_join(select(scrna_meta_tbl, sample, tumor_megasite, 
                   tumor_supersite, patient_id_short), 
            by = "sample") %>% 
  left_join(gois_tbl, by = c("gene", "cluster_label"))

## expression data for genes of interest per patient and (mega-)site
plot_data_summarized <- plot_data %>% 
  ## filter only for gene-cluster combinations of interest
  ## (no need to compute genes in clusters where they are not differential)
  filter(filter_column) %>% 
  group_by(gene, cluster_label, tumor_megasite, 
           tumor_supersite, patient_id_short) %>% 
  summarise(mean_value = mean(value, na.rm = T)) %>% 
  group_by(gene, cluster_label, tumor_megasite, patient_id_short) %>% 
  mutate(mean_value_mega = mean(mean_value, na.rm = T)) %>% 
  group_by(patient_id_short) %>% 
  filter(any(tumor_megasite %in% "Adnexa"), 
         any(tumor_megasite %in% "Other")) %>% 
  ungroup()

## wide table expression data to get within patient tendencies
plot_data_spread <- plot_data_summarized %>% 
  distinct(gene, cluster_label, tumor_megasite, patient_id_short, mean_value_mega) %>% 
  spread(tumor_megasite, mean_value_mega) %>% 
  mutate(delta_sign = case_when(
    Adnexa > Other ~ "pos",
    Adnexa < Other ~ "neg",
    T ~ "NA"
  )) %>% 
  group_by(gene, cluster_label) %>% 
  mutate(sum_pos = sum(delta_sign == "pos", na.rm = T), 
         sum_neg = -sum(delta_sign == "neg", na.rm = T)) %>% 
  replace_na(list(sum_pos = 0, sum_neg = 0)) %>% 
  mutate(delta_sum = sum_pos + sum_neg) %>% 
  ungroup() %>% 
  select(-sum_pos, -sum_neg)

plot_data_gathered <- plot_data_summarized %>% 
  left_join(plot_data_spread, by = c("gene", "cluster_label", "patient_id_short")) %>% 
  mutate(gene_cluster_label = paste0(gene, "\n", cluster_label)) %>% 
  arrange(delta_sum, tumor_megasite, cluster_label, gene, mean_value) %>% 
  mutate(rank_helper = paste0(gene, "-", cluster_label, "-", patient_id_short)) %>% 
  mutate(rank_helper = ordered(rank_helper, levels = unique(rank_helper))) %>% 
  mutate(rank_helper_format = rep(c(T, F), len = nrow(.))) %>% 
  mutate(rank_helper_label = ifelse(rank_helper_format, patient_id_short,
                                    paste0(patient_id_short, "       "))) %>% 
  mutate(gene_cluster_label = ordered(gene_cluster_label, levels = unique(gene_cluster_label))) %>% 
  mutate(vector_color = case_when(
    delta_sign == "pos" ~ "Other",
    delta_sign == "neg" ~ "Adnexa",
    T ~ "NA"
  ))

marker_tbl_delta_sum <- marker_tbl_sub %>% 
  left_join(plot_data_gathered %>% 
              distinct(gene, cluster_label, delta_sum),
            by = c("gene", "cluster_label")) %>% 
  filter(cluster_label %in% c("Cancer.cell.1", "Cancer.cell.2", "Cancer.cell.3")) %>% 
  group_by(cluster_label) %>% 
  mutate(gois = (delta_sum > 0 & avg_log2FC > 0) | (delta_sum < 0 & avg_log2FC < 0)) %>% 
  arrange(delta_sum)

```

## Volcano plot

```{r chunk_340_110, fig.width=6, fig.height=3.5}

volcano1 <- ggplot() +
  geom_point(aes(-avg_log2FC, -log10(p_val_adj)), 
             color = "grey80", size = 0.1,
             data = marker_tbl_delta_sum) +
  # geom_point(aes(-avg_log2FC, -log10(p_val_adj), size = abs(delta_sum)),
  #            color = "grey80",
  #            data = filter(marker_tbl_delta_sum, !is.na(delta_sum))) +
  geom_point(aes(-avg_log2FC, -log10(p_val_adj), color = cluster_label),
             data = filter(marker_tbl_delta_sum, gois) %>% arrange(-delta_sum) %>% slice(1)) +
  geom_text_repel(aes(-avg_log2FC, -log10(p_val_adj), label = gene),
                  max.overlaps = Inf, min.segment.length = 0.2,
                  # xlim = c(-Inf, Inf), ylim = c(0, Inf),
                  data = filter(marker_tbl_delta_sum, gois) %>% arrange(-delta_sum) %>% slice(1)) +
  geom_point(aes(-avg_log2FC, -log10(p_val_adj), color = cluster_label),
             data = filter(marker_tbl_delta_sum, gois) %>% slice(1)) +
  geom_text_repel(aes(-avg_log2FC, -log10(p_val_adj), label = gene),
                  # xlim = c(-Inf, Inf), ylim = c(0, Inf),
                  max.overlaps = Inf, min.segment.length = 0.2,
                  data = filter(marker_tbl_delta_sum, gois) %>% slice(1)) +
  scale_color_manual(values = clrs$cluster_label$Ovarian.cancer.super[c("Cancer.cell.1", "Cancer.cell.2", "Cancer.cell.3")]) +
  coord_cartesian(clip = "off") +
  scale_x_continuous() +
  scale_y_continuous() +
  labs(x = bquote(log[2] ~fold ~change),
       y = bquote(-log[10](q-value)),
       color = "Cluster")

volcano1

```

```{r chunk_340_120, fig.width=6, fig.height=3.5}

volcano2 <- ggplot() +
  geom_point(aes(-avg_log2FC, -log10(p_val_adj)), 
             color = "grey80", size = 0.1,
             data = marker_tbl_delta_sum) +
  geom_point(aes(-avg_log2FC, -log10(p_val_adj), color = cluster_label),
             data = filter(marker_tbl_delta_sum, gois, str_detect(gene, "IGF"))) +
  geom_text_repel(aes(-avg_log2FC, -log10(p_val_adj), label = gene),
                  max.overlaps = Inf, min.segment.length = 0,
                  data = filter(marker_tbl_delta_sum, gois, str_detect(gene, "IGF"))) +
  scale_color_manual(values = clrs$cluster_label$Ovarian.cancer.super) +
  coord_cartesian(clip = "off") +
  scale_x_continuous() +
  scale_y_continuous() +
  labs(x = bquote(~log[2] ~fold ~change),
       y = bquote(-log[10](q-value)),
       color = "Cluster")

volcano2

```

## Vector plots

```{r chunk_340_130, fig.width=10, fig.height=7}

plot_ectopic_genes <- function(cluster_oi, gois = "top_genes", 
                               ntop_genes = 5) {
  
  if (gois == "top_genes") {
    
    plot_data_sub <- plot_data_gathered %>% 
      group_by(cluster_label) %>% 
      filter(gene %in% unique(gene)[1:ntop_genes] | gene %in% rev(unique(gene))[1:ntop_genes]) %>% 
      filter(cluster_label == cluster_oi) %>% 
      ungroup()
    
  } else {
    
    ntop_genes <- length(gois)
    
    plot_data_sub <- plot_data_gathered %>% 
      group_by(cluster_label) %>% 
      filter(gene %in% gois) %>% 
      filter(cluster_label == cluster_oi) %>% 
      ungroup()
    
  }
  
  plot_data_vectors <- distinct(plot_data_sub, gene, cluster_label, patient_id_short, .keep_all = T)
  
  ggplot(plot_data_sub) +
    geom_point(aes(mean_value, rank_helper, shape = tumor_megasite)) +
    geom_segment(aes(x = Adnexa, xend = Other, 
                     y = rank_helper, yend = rank_helper, 
                     color = vector_color),
                 arrow = arrow(type = "open", length = unit(0.04, "npc")),
                 data = plot_data_vectors) +
    facet_wrap(~gene_cluster_label, scales = "free", ncol = ntop_genes) +
    scale_y_discrete(breaks = plot_data_sub$rank_helper, 
                     labels = plot_data_sub$rank_helper_label) +
    scale_color_manual(values = clrs$tumor_megasite[1:2], 
                       labels = c("Non-adnexa", "Adnexa")) +
    scale_shape_manual(values = shps$tumor_megasite) +
    theme(axis.title.y = element_blank()) +
    # remove_guides +
    labs(x = "Scaled expression", shape = "Tumor\nsite", color = "Higher in")
  
}

# plot_ectopic_genes("Cancer.cell.1")

cc_cluster_labels <- names(clrs$cluster_label$Ovarian.cancer.super)

ectopic_genes_plot_list <- lapply(cc_cluster_labels, plot_ectopic_genes) %>% 
  setNames(cc_cluster_labels)

for (i in 1:length(ectopic_genes_plot_list)) {
  
  ectopic_genes_plot_list[[i]]  
  ggsave(paste0("figures/340_cancer_cell_ectopic_expression/340_ectopic_expression_", cc_cluster_labels[i], ".pdf"), ectopic_genes_plot_list[[i]], width = 13, height = 7)
  ggsave(paste0("figures/340_cancer_cell_ectopic_expression/340_ectopic_expression_", cc_cluster_labels[i], ".png"), ectopic_genes_plot_list[[i]], width = 13, height = 7)
  
}

```

```{r chunk_340_140, fig.width=4, fig.height=3.5}

vector1 <- plot_ectopic_genes("Cancer.cell.1", "IGFBP7")
# vector1 <- plot_ectopic_genes("Cancer.cell.4", "PEG10")
# vector1 <- plot_ectopic_genes("Cancer.cell.2", "APOE")
vector1

```

## arranged grid

```{r chunk_340_160, fig.width=10, fig.height=3.5}

ggdraw() +
  draw_plot(volcano1, height = 1, width = 0.6, x = 0, y = 0) + 
  draw_plot(vector1, height = 1, width = 0.4, x = 0.6, y = 0)

ggsave("figures/340_cancer_cell_ectopic_expression/340_IGFBP7_grid.pdf", width = 10, height = 3.5)

```

## session info 

```{r chunk_340_999}

devtools::session_info()

```

